### PR TITLE
[MM-52973] Avoid thundering herd problem in IsFirstUserAccount

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
https://github.com/mattermost/mattermost-server/pull/23545 already greatly improved the performance of `IsFirstUserAccount` by not re-fetching the number of users from the store if more than one is found.

This PR improves the performance even more by fixing the thundering herd problem in `IsFirstUserAccount`. On a server start, a lot of users might try to connect. That causes a lot of concurrent calls to `IsFirstUserAccount`. This PR ensure that the DB is only hit once and not multiple times.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-52973

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/channels/app/platform/service.go [hunk 1]
```diff
@@ -49,11 +49,13 @@ type PlatformService struct {
...
-	clientConfigHash                       atomic.Value
-	limitedClientConfig                    atomic.Value
-	fetchUserCountForFirstUserAccountCheck atomic.Bool
+	asymmetricSigningKey atomic.Value
+	clientConfig         atomic.Value
+	clientConfigHash     atomic.Value
+	limitedClientConfig  atomic.Value
+
+	isFirstUserAccountLock sync.Mutex
+	isFirstUserAccount     atomic.Bool
```
Comment (hanzei) [inline server/channels/app/platform/service.go:58]: 2/5 that `isFirstUserAccount` is easier to understand

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

**Review by BenCookie95 — APPROVED** [review]

**Comment by hanzei** [comment]
Note to QA: The e2e-tests `MM-T2500_2 "X" closes the dialog` and `MM-T470 Reply in existing GM` failed for this PR. I've fairly certain that it's unrelated to the changes, as I've seen that test fail for other PRs.

**Pull request merged**